src/main/java/org/bytedeco/javacpp/tools/Parser.java

```diff
@@ -2512,8 +2517,34 @@ boolean function(Context context, DeclarationList declList) throws ParserExcepti
                 tokens.next();
             }
 
+            /* If it's a friend function, and we want friend mapping, check if it accepts an argument of the enclosing type.
+               If it does, add a Java instance method calling the static native method.
+               If it does not, skip over. */
+            Info info2 = infoMap.getFirst(null);
+            boolean mapFriends = info != null ? info.mapFriends : info2 != null ? info2.mapFriends : false;
+            if (type.friend && mapFriends) {
+                Declarator[] paramDeclarators = dcl.parameters.declarators;
+                String argList = "", staticArgList = "";
+                boolean foundThis = false;
+                for (Declarator paramDecl: paramDeclarators) {
+                    if (staticArgList.length() > 0) staticArgList += ", ";
+                    if (!foundThis && paramDecl.type.cppName.equals(context.cppName)) {
+                        foundThis = true;
+                        staticArgList += "this";
+                    } else {
+                        if (argList.length() > 0) argList += ", ";
+                        argList += paramDecl.type.javaName+' '+paramDecl.javaName;
+                        staticArgList += paramDecl.javaName;
+                    }
+                }
+                if (foundThis)
+                    decl.text += "public "+dcl.type.javaName+" "+dcl.javaName+"("+argList+") { "+(dcl.type.javaName.equals("void") ? "" : "return ")+dcl.javaName+"("+staticArgList+"); }\n";
+                else
+                    mapFriends = false;
+            }
+
             // skip over non-const function within const class
-            if (!decl.constMember && context.constName != null) {
+            if (type.friend && !mapFriends || !decl.constMember && context.constName != null) {
```

Comment:
To make sure we don't have any regressions, let's keep the skip condition for friend above instead.

Reply:
I've pushed proposed changes in that direction. Can you give it a try? If it works, I think this is good to merge.